### PR TITLE
SMTP App: Tests for main business useCase

### PR DESCRIPTION
**Pull request by lkostrowski**
## Scope of the PR

todo

- [x] finish tests for successful response

- [x] convert errors from useCase to response status codes

## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/smtp/src/pages/api/webhooks/gift-card-sent.ts [hunk 1]
```diff
@@ -103,14 +105,44 @@ const handler: NextWebhookApiHandler<GiftCardSentWebhookPayloadFragment> = async
...
+      result.match(
+        (r) => {
+          logger.info("Successfully sent email(s)");
+
+          return res.status(200).json({ message: "The event has been handled" });
+        },
+        (err) => {
+          const errorInstance = err[0];
+
+          if (errorInstance instanceof SendEventMessagesUseCase.ServerError) {
```
Comment (krzysztofzuraw) [inline apps/smtp/src/pages/api/webhooks/gift-card-sent.ts:125]: **Question:** can't you use here switch with error constructor?

**Review by krzysztofzuraw — COMMENTED** [review]
Looks good - one readability comment

File: apps/smtp/src/pages/api/webhooks/gift-card-sent.ts (hunk 1 above)
Comment (lkostrowski) [reply apps/smtp/src/pages/api/webhooks/gift-card-sent.ts:125]: Not really due to prototype chain. Here is demo:

CleanShot 2024-05-20 at 09 48 58

**Review by krzysztofzuraw — APPROVED** [review]

File: apps/smtp/src/pages/api/webhooks/invoice-sent.ts [hunk 2]
```diff
@@ -84,14 +86,44 @@ const handler: NextWebhookApiHandler<InvoiceSentWebhookPayloadFragment> = async
...
+          if (errorInstance instanceof SendEventMessagesUseCase.ServerError) {
+            logger.error("Failed to send email(s) [server error]", { error: err });
+
+            return res.status(500).json({ message: "Failed to send email" });
+          } else if (errorInstance instanceof SendEventMessagesUseCase.ClientError) {
+            logger.info("Failed to send email(s) [client error]", { error: err });
+
+            return res.status(400).json({ message: "Failed to send email" });
+          } else if (errorInstance instanceof SendEventMessagesUseCase.NoOpError) {
+            logger.error("Sending emails aborted [no op]", { error: err });
```
Comment (witoszekdev) [inline apps/smtp/src/pages/api/webhooks/invoice-sent.ts:115]: question: Should we actually log error here? NoOp error is when settings are missing or config is not active. We don't really have control over this.

```ts
  static NoOpError = BaseError.subclass("SendEventMessagesUseCaseServerError");

  static EventConfigNotActiveError = this.NoOpError.subclass("EventConfigNotActiveError");

  static EventSettingsMissingError = this.NoOpError.subclass("EventSettingsMissingError");
```

**Review by witoszekdev — APPROVED** [review]

File: apps/smtp/src/pages/api/webhooks/invoice-sent.ts (hunk 2 above)
Comment (lkostrowski) [reply apps/smtp/src/pages/api/webhooks/invoice-sent.ts:115]: Yes, you are right. This should be normal log 👍

**Pull request merged**